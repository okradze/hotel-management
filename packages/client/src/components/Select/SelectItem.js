import React from 'react'

const SelectItem = ({ setIsOpen, setValue, text, value }) => (
    <li className="select__item">
        <button
            onKeyPress={e => {
                if (e.which === 13) {
                    setValue(text, value)
                    setIsOpen(false)
                }
            }}
            onClick={() => {
                setValue(text, value)
                setIsOpen(false)
            }}
            className="button--transparent"
            type="button"
        >
            {text}
        </button>
    </li>
)

export default SelectItem
