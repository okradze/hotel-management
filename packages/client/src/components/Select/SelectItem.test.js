import SelectItem from './SelectItem'

describe('SelectItem', () => {
    let wrapper, setIsOpen, setValue, text, value

    beforeEach(() => {
        setIsOpen = jest.fn()
        setValue = jest.fn()
        text = 'abc'
        value = 'abc123'
        wrapper = shallow(
            <SelectItem
                setIsOpen={setIsOpen}
                setValue={setValue}
                text={text}
                value={value}
            />,
        )
    })
    afterEach(() => {
        jest.clearAllMocks()
    })

    it('renders', () => {
        expect(wrapper).toMatchSnapshot()
    })

    describe('calls setValue and setIsOpen on', () => {
        it('click', () => {
            wrapper.find('button').simulate('click')
            expect(setValue).toHaveBeenCalledWith(text, value)
            expect(setIsOpen).toHaveBeenCalledWith(false)
        })

        it('keypress', () => {
            wrapper.find('button').simulate('keypress', {
                which: 13,
            })
            expect(setValue).toHaveBeenCalledWith(text, value)
            expect(setIsOpen).toHaveBeenCalledWith(false)
        })
    })
})
