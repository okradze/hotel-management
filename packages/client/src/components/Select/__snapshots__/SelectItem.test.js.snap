// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectItem renders 1`] = `
<li
  className="select__item"
>
  <button
    className="button--transparent"
    onClick={[Function]}
    onKeyPress={[Function]}
    type="button"
  >
    abc
  </button>
</li>
`;
